src/integer_mod_q/mat_zq/arithmetic/mul.rs

```diff
@@ -61,8 +60,7 @@ impl Mul<&MatZ> for &MatZq {
     type Output = MatZq;
 
     /// Implements the [`Mul`] trait for [`MatZq`] and [`MatZ`].
-    ///
-    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatZq`] and [`MatZ`].
+    /// [`Mul`] is implemented for any combination of owned and borrowed values.
```

Comment:
sometimes you sticked to any combination of owned and borrowed and sometimes you provided them implicitely. Any specific reason?

Reply:
Yes, on the same types, we name them explicitly and on different types, we name them implicitly